astropy/io/fits/hdu/image.py

```diff
@@ -264,19 +264,16 @@ def data(self, data):
             self._data_replaced = True
             was_unsigned = False
 
-        if (
-            data is not None
-            and not isinstance(data, np.ndarray)
-            and not _is_dask_array(data)
-        ):
-            # Try to coerce the data into a numpy array--this will work, on
-            # some level, for most objects
-            try:
-                data = np.array(data)
-            except Exception:
-                raise TypeError(
-                    f"data object {data!r} could not be coerced into an ndarray"
-                )
+        if data is not None:
+            if not isinstance(data, np.ndarray) and not _is_dask_array(data):
+                # Try to coerce the data into a numpy array--this will work, on
+                # some level, for most objects
+                try:
+                    data = np.array(data)
+                except Exception:
+                    raise TypeError(
```

Comment:
Coverage is complaining about this TypeError, even though I did not change this part. That said it is right in the sense that this TypeError is indeed never checked in any test.
More importantly: I can't think of any scenario where "data = np.array(data)" would actually raise an exception. Numpy seems to happily eat anything you throw at it, and if it is not array_like it will just create a 0-dimensional array containing the object. So I believe this part of the code is actually not needed. I will remove it.

Reply:
In this case, you may use `# pragma: no cover` to ask coverage to stop complaining. The check goes through diff even though you might not have really touched that code.

Reply:
I think I would rather delete the try-except, assuming there's no way it will trigger (I don't think it should be possible, looking over at the documentation).

Reply:
Fixed in the latest commit, PR should be ready now.